iceoryx_examples/iceperf/README.md

```diff
@@ -175,127 +177,181 @@ Here an example output with Ubuntu 18.04 on Intel(R) Xeon(R) CPU E3-1505M v5 @ 2
+    std::cout << std::endl;
+    std::cout << "Finished!" << std::endl;
+}
+```
 
-In the `main()` method we create instances for the different IPC technologies we want to compare. Each one is implemented in an own class and implements the pure virtual functions provided with the `IcePerfBase` class
+Initialization is different for each IPC technology. Here we have to create sockets, message queues or iceoryx publisher and subscriber.
+With `ipcTechnology.initLeader()` we are setting up these resources on the leader side.
+After the definition of the different payload sizes to use, we execute a single round trip measurement for each individual payload size.
+The leader has to orchestrate the whole process and has a pre and post step for each ping pong round trip measurement.
+`ipcTechnology.prePingPongLeader()` sets the payload size for the upcoming measurement.
+`ipcTechnology.pingPongLeader(numRoundtrips)` then does the ping pong between leader and follower and returns
+the time it took to do the provided number of round trips.  After the measurements were done for all the different payload sizes,
+`ipcTechnology.releaseFollower()` releases the follower that is not aware of things like how many payload sizes are considered.
+After cleaning up the communication resources with `ipcTechnology.shutdown()` the results are printed.
 
-iceperf-laurel, the leader in this setup, takes the number of round trips to perform for each payload size as command line parameter. We check if one was provided and take this or otherwise use the default one. The higher the number of round trips, the more accurate the measurements will be but you have to consider that an iceperf run can take quite a long time then.   
+The `followerDo()` method is much simpler as the follower only reacts and does not the control. Besides `ipcTechnology.initFollower()` and `ipcTechnology.shutdown()` all the functionality to do the ping pong for different payload sizes is done in `ipcTechnology.pingPongFollower()`
```

Comment:
... the follower, it reacts only and does not control. (or does not orchestrate)